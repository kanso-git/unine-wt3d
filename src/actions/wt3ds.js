
// ADD_WT3D
export const addWt3d = (expense) => ({
  type: 'ADD_WT3D',
  expense
});

export const startAddWt3d = (expenseData = {}) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    const {
      description = '',
      note = '',
      amount = 0,
      createdAt = 0
    } = expenseData;
    const expense = { description, note, amount, createdAt };

    return database.ref(`users/${uid}/wt3ds`).push(expense).then((ref) => {
      dispatch(addWt3d({
        id: ref.key,
        ...expense
      }));
    });
  };
};

// REMOVE_WT3D
export const removeWt3d = ({ id } = {}) => ({
  type: 'REMOVE_WT3D',
  id
});

export const startRemoveWt3d = ({ id } = {}) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    return database.ref(`users/${uid}/wt3ds/${id}`).remove().then(() => {
      dispatch(removeWt3d({ id }));
    });
  };
};

// EDIT_WT3D
export const editWt3d = (id, updates) => ({
  type: 'EDIT_WT3D',
  id,
  updates
});

export const startEditWt3d = (id, updates) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    return database.ref(`users/${uid}/wt3ds/${id}`).update(updates).then(() => {
      dispatch(editWt3d(id, updates));
    });
  };
};

// SET_WT3DS
export const setWt3ds = (wt3ds) => ({
  type: 'SET_WT3DS',
  wt3ds
});

export const startSetWt3ds = () => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    return database.ref(`users/${uid}/wt3ds`).once('value').then((snapshot) => {
      const wt3ds = [];

      snapshot.forEach((childSnapshot) => {
        wt3ds.push({
          id: childSnapshot.key,
          ...childSnapshot.val()
        });
      });

      dispatch(setWt3ds(wt3ds));
    });
  };
};
