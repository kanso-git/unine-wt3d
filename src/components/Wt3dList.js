import React from 'react';
import { connect } from 'react-redux';
import Wt3dListItem from './Wt3dListItem';
import selectWt3ds from '../selectors/wt3ds';

export const Wt3dList = props => (
  <div className="content-container">
    <div className="list-header">
      <div className="show-for-mobile">wt3ds</div>
      <div className="show-for-desktop">Wt3d</div>
      <div className="show-for-desktop">Amount</div>
    </div>
    <div className="list-body">
      {
        props.wt3ds.length === 0 ? (
          <div className="list-item list-item--message">
            <span>No wt3ds</span>
          </div>
        ) : (
            props.wt3ds.map(wt3d => <Wt3dListItem key={wt3d.id} {...wt3d} />)
          )
      }
    </div>
  </div>
);

const mapStateToProps = state => ({
  wt3ds: selectWt3ds(state.wt3ds, state.filters),
});

export default connect(mapStateToProps)(Wt3dList);
