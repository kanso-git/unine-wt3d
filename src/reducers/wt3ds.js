// wt3ds Reducer

const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_WT3D':
      return [
        ...state,
        action.expense
      ];
    case 'REMOVE_WT3D':
      return state.filter(({ id }) => id !== action.id);
    case 'EDIT_WT3D':
      return state.map((expense) => {
        if (expense.id === action.id) {
          return {
            ...expense,
            ...action.updates
          };
        } else {
          return expense;
        };
      });
    case 'SET_WT3DS':
      return action.wt3ds;
    default:
      return state;
  }
};
