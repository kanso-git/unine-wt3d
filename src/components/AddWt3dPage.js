import React from 'react';
import { connect } from 'react-redux';
import Wt3dForm from './Wt3dForm';
import { startAddWt3d } from '../actions/wt3ds';


export class AddWt3dPage extends React.Component {
  onSubmit = (expense) => {
    this.props.startAddWt3d(expense);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add new waytouch</h1>
          </div>
        </div>
        <div className="content-container">
          <Wt3dForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startAddWt3d: expense => dispatch(startAddWt3d(expense)),

});


export default connect(undefined, mapDispatchToProps)(AddWt3dPage);
