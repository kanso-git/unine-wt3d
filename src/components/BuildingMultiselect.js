import React, { Component } from 'react';
import { connect } from 'react-redux';
import 'react-select/dist/react-select.css';
import Select from 'react-select';

import { bdl } from '../actions';

const staticdata = [
  { label: 'Chocolate', value: 'chocolate' },
  { label: 'Vanilla', value: 'vanilla' },
  { label: 'Strawberry', value: 'strawberry' },
  { label: 'Caramel', value: 'caramel' },
  { label: 'Cookies and Cream', value: 'cookiescream' },
  { label: 'Peppermint', value: 'peppermint' },
];

const WHY_WOULD_YOU = [
  { label: 'Chocolate (are you crazy?)', value: 'chocolate', disabled: true },
].concat(staticdata.slice(1));

class BuildingMultiselect extends Component {

  state={
    removeSelected: true,
    disabled: false,
    crazy: false,
    stayOpen: false || this.props.stayOpen,
    value: [],
    rtl: false,
  }
  componentDidMount() {
    this.getBuildingList();
  }
  getBuildingList = async () => {
    await this.props.getBuildingList();
  }
  handleSelectChange= (value) => {
    console.log('You\'ve selected:', value);
    this.setState({ value });
    this.props.handleSelect(value);
  }

  toggleCheckbox = (e) => {
    this.setState({
      [e.target.name]: e.target.checked,
    });
  }

  toggleRtl= (e) => {
    const rtl = e.target.checked;
    this.setState({ rtl });
  }

  render() {
    const {
      crazy, disabled, stayOpen, value,
    } = this.state;
    const options = crazy ? WHY_WOULD_YOU : this.props.buildingList;
    return (
      <div className="section">
        <h3 className="section-heading">{this.props.label} </h3>
        <Select
          closeOnSelect={!stayOpen}
          disabled={disabled}
          multi
          onChange={this.handleSelectChange}
          options={options}
          placeholder="Select your building(s)"
          removeSelected={this.state.removeSelected}
          rtl={this.state.rtl}
          simpleValue
          value={value}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  const buildingList = state.bdl.buildingList;
  return { buildingList };
};

export default connect(mapStateToProps,{...bdl})(BuildingMultiselect);
