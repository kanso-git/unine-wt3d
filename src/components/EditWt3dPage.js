import React from 'react';
import { connect } from 'react-redux';
import Wt3dForm from './Wt3dForm';
import { startEditWt3d, startRemoveWt3d } from '../actions/wt3ds';

export class EditWt3dPage extends React.Component {
  onSubmit = (expense) => {
    this.props.startEditWt3d(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.startRemoveWt3d({ id: this.props.expense.id });
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Wt3d</h1>
          </div>
        </div>
        <div className="content-container">
          <Wt3dForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button className="button button--secondary" onClick={this.onRemove}>Remove Wt3d</button>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  expense: state.wt3ds.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  startEditWt3d: (id, expense) => dispatch(startEditWt3d(id, expense)),
  startRemoveWt3d: (data) => dispatch(startRemoveWt3d(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditWt3dPage);
