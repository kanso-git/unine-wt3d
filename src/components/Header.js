import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import { faPowerOff } from '@fortawesome/fontawesome-free-solid';

import { startLogout } from '../actions/auth';

export const Header = ({ startLogout, auth }) => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/dashboard">
          <h1>Waytouch 3D</h1>
        </Link>
        <button className="button button--link" onClick={startLogout}>
          <FontAwesomeIcon icon={faPowerOff} /> Logout ({auth.user.displayName})
        </button>
      </div>
    </div>
  </header>
);

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout()),
});

const mapStateToProps = state => ({
  auth: state.auth,
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
