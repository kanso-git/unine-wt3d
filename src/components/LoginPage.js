import React, { Component } from 'react';
import { connect } from 'react-redux';
import { authActions } from '../actions';

export class LoginPage extends Component {
  state={
    error: null,
    username: '',
    password: '',
  }
  
  onUsernameChange = (e) => {
    const username = e.target.value;
    this.setState(() => ({ username }));
  }
  onPasswordChange = (e) => {
    const password = e.target.value;
    this.setState(() => ({ password }));
  }
  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.username || !this.state.password) {
      this.setState(() => ({ error: 'Please provide username and password.' }));
    } else {
      this.setState(() => ({ error: '' }));
      this.props.startLogin(this.state.username, this.state.password);
      e.target.reset();
    }
  };

  render() {
    return (
      <div className="box-layout">
        <div className="box-layout__box">
          <h1 className="box-layout__title">Waytouch 3D</h1>
          <form className="form" onSubmit={this.onSubmit} >
            {this.state.error && <p className="form__error">{this.state.error}</p>}
            <input
              type="text"
              placeholder="Login"
              autoComplete="off"
              className="text-input"
              value={this.state.username}
              onChange={this.onUsernameChange}
            />
            <input
              type="password"
              placeholder="Password"
              className="text-input"
              autoComplete="off"
              value={this.state.password}
              onChange={this.onPasswordChange}
            />

            <p> It's time to get your Waytouch under control.</p>
            <button className="button" onClick={this.login} >Login</button>
          </form>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => (
  {
    auth: state.auth,
  });


export default connect(mapStateToProps, { ...authActions })(LoginPage);
