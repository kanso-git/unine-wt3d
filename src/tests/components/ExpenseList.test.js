import React from 'react';
import { shallow } from 'enzyme';
import { Wt3dList } from '../../components/Wt3dList';
import wt3ds from '../fixtures/wt3ds';

test('should render Wt3dList with wt3ds', () => {
  const wrapper = shallow(<Wt3dList wt3ds={wt3ds} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render Wt3dList with empty message', () => {
  const wrapper = shallow(<Wt3dList wt3ds={[]} />);
  expect(wrapper).toMatchSnapshot();
});
