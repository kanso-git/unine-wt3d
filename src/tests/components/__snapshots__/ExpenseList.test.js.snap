// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Wt3dList with empty message 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      wt3ds
    </div>
    <div
      className="show-for-desktop"
    >
      Wt3d
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <div
      className="list-item list-item--message"
    >
      <span>
        No wt3ds
      </span>
    </div>
  </div>
</div>
`;

exports[`should render Wt3dList with wt3ds 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      wt3ds
    </div>
    <div
      className="show-for-desktop"
    >
      Wt3d
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <Wt3dListItem
      amount={195}
      createdAt={0}
      description="Gum"
      id="1"
      note=""
    />
    <Wt3dListItem
      amount={109500}
      createdAt={-345600000}
      description="Rent"
      id="2"
      note=""
    />
    <Wt3dListItem
      amount={4500}
      createdAt={345600000}
      description="Credit Card"
      id="3"
      note=""
    />
  </div>
</div>
`;
